dftimewolf/lib/exporters/timesketch.py

```diff
@@ -19,37 +19,43 @@ class TimesketchExporter(BaseExporter):
     sketch_id: Timesketch Sketch ID the timeline will be added to
   """
 
-  def __init__(self, timesketch_api, incident_id, sketch_id, verbose, paths):
+  def __init__(self, auth, incident_id, sketch_id, verbose, paths):
     """Initializes a filesystem collector.
 
     Args:
-      timesketch_api: Timesketch API object
+      auth: an (endpoint, username, password) tuple
       incident_id: Incident ID or description associated with the investigation
       sketch_id: If provided, append the timelines to a given sketch
       verbose: Whether verbose output is desired.
       paths: List of (name, path) tuples to export
     """
     super(TimesketchExporter, self).__init__(verbose=verbose)
-
-    self.timesketch_api = timesketch_api
-    self.incident_id = incident_id
-    self.sketch_id = sketch_id
-    self.paths = paths
-    self.sketch_url = None
-    self.output = None
-
-    if not self.sketch_id:
-      if incident_id:
-        sketch_name = incident_id
-        sketch_description = incident_id
-      else:
-        sketch_name = 'Untitled sketch'
-        sketch_description = 'No description provided'
-      self.sketch_id = timesketch_api.create_sketch(
-          sketch_name, sketch_description)
-      self.console_out.StdOut(
-          'New sketch created: {0:d}'.format(self.sketch_id))
-      syslog.syslog('New sketch created: {0:d}'.format(self.sketch_id))
+    ts_endpoint, ts_username, ts_password = auth
+    self.timesketch_api = timesketch_utils.TimesketchApiClient(
+        ts_endpoint, ts_username, ts_password)
+    if not self.timesketch_api.session:
+      self.errors.append("Could not connect to timesketch server {0:s}".format(
+          ts_endpoint
+      ))
```

Comment:
Move these to the line above. And don't you want to fail here? Just return, and remove the "else" etc.